userProfile/src/Components/Form.tsx

```diff
@@ -0,0 +1,155 @@
+    city: ''
+};
+
+const defaultData: FormData = {
+    name: '',
+    surname: '',
+    city: cities[0]
+};
+
+const Form = () => {
+    const [name, setName] = useState(defaultData.name);
+    const [surname, setSurname] = useState(defaultData.surname);
+    const [city, setCity] = useState(defaultData.city);
+
+    const [person, setPerson] = useState({
+        name,
+        surname,
+        city
+    });
+
+    const [opened, setOpened] = useState(false);
+
+    const handlerField = (field: keyof FormData, value: string) => {
+        switch (field) {
+            case 'name':
+                setName(value);
+                break;
+            case 'surname':
+                setSurname(value);
+                break;
+            case 'city':
+                setCity(value);
+                break;
+            default:
+                throw new Error();
+        }
+    };
+
+    const submitForm = () => {
+        setPerson({ name: name, surname: surname, city: city });
+        setOpened(true);
+    };
+
+    const renderForm = () => {
+        return (
+            <form className="form">
+                <Gapped gap={15} vertical>
+                    <label>
+                        <div className="label">Имя</div>
+                        <Input
+                            placeholder="Введите имя пользователя"
+                            value={name}
+                            onChange={e => handlerField('name', e.target.value)}
+                        />
+                    </label>
+                    <label>
+                        <div className="label">Фамилия</div>
+                        <Input
+                            placeholder="Введите имя пользователя"
+                            value={surname}
+                            onChange={e => handlerField('surname', e.target.value)}
+                        />
+                    </label>
+                    <label>
+                        <div className="label">Город</div>
+                        <Select<string>
+                            placeholder="Выберите город"
+                            items={cities}
+                            value={city}
+                            onValueChange={e => handlerField('city', e)}
+                        />
+                    </label>
+
+                    <Button use="primary" size="large" onClick={submitForm}>
+                        Сохранить
+                    </Button>
+                </Gapped>
+            </form>
+        );
+    };
+
+    const renderModal = () => {
+        const closeModal = () => {
+            saveData.name = name;
+            saveData.surname = surname;
+            saveData.city = city;
+
+            setName(defaultData.name);
+            setSurname(defaultData.surname);
+            setOpened(false);
+        };
+
+        const renderDiff = (field: keyof FormData, fieldName: string) => {
+            if (saveData[field] !== person[field] && !isNothingChanged) {
```

Comment:
Зачем здесь проверка на то, изменилось ли что-то в форме или нет? Если мы дошли до этой функции, то `!isNothingChanged` всегда будет равен `true`. Но даже если представить, что ранее этой проверки не было, тогда нам всё равно не важно есть ли изменения во всей форме, нам важно только изменение в конкретном поле + если рассмотреть все варианты (построить таблицу истинности этого выражения), станет понятно, что его можно упростить
```suggestion
            if (saveData[field] !== person[field]) {
```